packages/php-wasm/web-service-worker/src/worker-caching.ts

```diff
@@ -0,0 +1,86 @@
+// import { logger } from '@php-wasm/logger';
+import { isURLScoped } from '@php-wasm/scopes';
+
+const wpCacheKey = 'playground-cache';
+
+const isCacheFull = async () => {
+	if (navigator.storage && navigator.storage.estimate) {
+		const estimatedStorage = await navigator.storage.estimate();
+		if (estimatedStorage.usage && estimatedStorage.quota) {
+			return estimatedStorage.usage >= estimatedStorage.quota;
+		}
+	}
+	return false;
+};
+
+export const addCache = async (key: string, response: Response) => {
+	if (await isCacheFull()) {
+		console.warn('Cache is full, not caching response');
+		return;
+	}
+	try {
+		const clonedResponse = response.clone();
+		const cache = await caches.open(wpCacheKey);
+		await cache.put(key, clonedResponse);
+	} catch (e) {
+		console.warn('Failed to cache response', e);
+	}
+};
+
+export const getCache = async (key: string) => {
+	const cache = caches.open(wpCacheKey);
+	return await cache.then((c) => c.match(key));
+};
+
+const validHostnames = ['playground.wordpress.net', '127.0.0.1', 'localhost'];
```

Comment:
Nit: `validHostname` suggests a string that a URL parser would recognize as a hostname. What do you think about `cacheEnabledHosts`?

Reply:
Also, could this be inlined into the `isValidHostname` function below?

Reply:
Moved it inline